File: query.lua
local http = require("socket.http")
local ltn12 = require("ltn12")

-- Define the request body
local request_body = "host=<hostname>&username=<username>&password=<password>&query=<query>"

-- Define the request options
local request_options = {
  method = "POST",
  headers = {
    ["Content-Type"] = "application/x-www-form-urlencoded",
    ["Content-Length"] = #request_body
  },
  source = ltn12.source.string(request_body),
  sink = ltn12.sink.table(response)
}

-- Send the request and receive the response
local response_code, response_headers, response = http.request("http://<php_file_url>", request_options)

-- If the request is successful, parse the JSON object
if response_code == 200 then
  local json = require("cjson")
  local response_data = json.decode(table.concat(response))
  -- Do something with the response data
else
  -- If the request fails, handle the error
  error("Request failed with code " .. response_code)
end

File: query.php
<?php

header("Content-Type: application/json");

if ($_SERVER['REQUEST_METHOD'] === 'POST') {
    $host = $_POST['host'];
    $username = $_POST['username'];
    $password = $_POST['password'];
    $query = $_POST['query'];
    $dbname = $_POST['dbname'];
    
    // Connect to host server
    $conn = mysqli_connect($host, $username, $password, $dbname);
    
    if (!$conn) {
        // If connection fails, return an error message
        $response = array("error" => "Unable to connect to host server");
        echo json_encode($response);
        exit;
    }
    
    // Execute the query
    $result = mysqli_query($conn, $query);
    
    if (!$result) {
        // If query fails, return an error message
        $response = array("error" => "Unable to execute the query");
        echo json_encode($response);
        exit;
    }
    
    // If query succeeds, return the result as a JSON object
    $response = array();
    while ($row = mysqli_fetch_assoc($result)) {
        $response[] = $row;
    }
    echo json_encode($response);
    exit;
}

?>
